src/pages/index.js

```diff
@@ -2,76 +2,226 @@ import "./index.css";
 const jobInput = document.querySelector("#profile-description-input");
 const cardAddForm = addModal.querySelector("#add-modal-form");
-const cardListEl = document.querySelector(".cards__list");
+const profilePicture = document.querySelector("#profile-change-button");
+const pictureChangeModal = document.querySelector("#profile-change-picture");
+const pictureChangeForm = pictureChangeModal.querySelector(
+  "#picture-modal-form"
+);
+const deleteFormButton = document.querySelector(".cards__delete-button");
 
 const cardTemplateSelector = "#card-template";
 
-const editCardModal = new PopupWithForm(
-  "#profile-edit-modal",
-  handleProfileEditSubmit
-);
-editCardModal.setEventListeners();
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "6a75e942-1cdf-4570-be93-f1316f67106a",
+    "Content-Type": "application/json",
+  },
+});
 
-const previewImageModal = new PopupWithImage("#preview-image-modal");
-previewImageModal.setEventListeners();
-const userInfo = new UserInfo(".profile__title", ".profile__description");
-const itemSection = new Section({
-  items: initialCards,
-  renderer: renderCard,
-  cssSelector: ".cards__list",
+let itemSection;
+
+api.getCardList().then((items) => {
+  itemSection = new Section({
+    items,
+    renderer: renderCard,
+    cssSelector: ".cards__list",
+  });
+
+  itemSection.renderItems(items);
 });
-itemSection.renderItems();
 
-const addCardModal = new PopupWithForm(
-  "#profile-add-card",
-  handleCardAddSubmit
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
 );
-addCardModal.setEventListeners();
 
-function handleImageClick(name, link) {
-  previewImageModal.open(name, link);
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo(data);
+    // a = true
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      deleteConfirm.renderDelete(true);
+      deleteConfirm.close();
+      card.remove();
+    })
+    .catch(() => {
+      deleteConfirm.renderDelete(false);
```

Comment:
You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

**Please, correct it everywhere**